pints/_mcmc/__init__.py

```diff
@@ -1100,3 +1100,80 @@ def mcmc_sample(log_pdf, chains, x0, sigma0=None, method=None):
     """
     return MCMCController(    # pragma: no cover
         log_pdf, chains, x0, sigma0, method=method).run()
+
+
+def sample_initial_points(log_pdf, n_points, random_sampler=None,
+                          max_tries=None, parallel=False, n_workers=None):
+    """
+    Draws parameter values from a given sampling distribution until either
+    finite values for each of ``n_points`` have been generated or the total
+    number of attempts exceeds ``max_tries``.
+
+    If ``log_pdf`` is of :class:`LogPosterior`, then the
+    ``log_pdf.log_prior().sample`` method is used for initialisation, although
+    this is overruled by ``random_sampler`` if it is supplied.
```

Comment:
```suggestion
    Samples ``n_points`` parameter values to use as starting points in a
    sampling or optimisation routine on the given ``log_pdf``.
    
    If a method ``random_sampler`` is provided, then this will be used to draw 
    the random samples. If no sampler method is given but ``log_pdf`` is a
    :class:`LogPosterior`, then the method ``log_pdf.log_prior().sample()``
    will be used. A ``ValueError`` is raised if neither option is available.
    
    Each sample ``x`` is tested to ensure that ``log_pdf(x)`` returns a finite
    result. If not, a new sample will be drawn. This is repeated at most
    ``max_tries`` times, after which an error is raised.
```

Reply:
I've taken large parts of this and incorporated into what I have now.